File: lang/c/test/vac.fun/algorithms/search/LinearSearchTest.c
#include <vac.fun/algorithms/search/LinearSearch.h>
#include <vac.fun/util/SequenceBuilder.h>
#include <vac.fun/util/TestRunner.h>

bool testLinearSearch(void) {
    size_t size = 32768;

    int arr[size];
    SequenceBuilder_packIncreasing(arr, size);

    if (LinearSearch_find(arr, size, -1) != size) {
        return false;
    }

    if (LinearSearch_find(arr, size, 2147483647) != size) {
        return false;
    }

    for (size_t i = 0; i < size; i++) {
        if (LinearSearch_find(arr, size, arr[i]) != i) {
            return false;
        }
    }

    return true;
}

int main(void) {
    TestRunner_parseTest(testLinearSearch());
}

File: lang/nodejs/test/vac.fun/algorithms/search/LinearSearchTest.js
"use strict"

const LinearSearch    = require("vac.fun/algorithms/search/LinearSearch")
const SequenceBuilder = require("vac.fun/util/SequenceBuilder")
const TestRunner      = require("vac.fun/util/TestRunner")

const testLinearSearch = () => {
    let size = 32768

    const arr = new Array(size)
    SequenceBuilder.packIncreasing(arr)

    if (LinearSearch.find(arr, -1) !== size) {
        return false
    }

    if (LinearSearch.find(arr, 2147483647) !== size) {
        return false
    }

    for (let i = 0; i < size; i++) {
        if (LinearSearch.find(arr, arr[i]) !== i) {
            return false
        }
    }

    return true
}

if (module === require.main) {
    TestRunner.parseTest(testLinearSearch())
}
